internal/vault/cache.go

```diff
@@ -133,6 +134,14 @@ func (c *clientCache) Prune(filterFunc ClientCachePruneFilterFunc) []ClientCache
 	return pruned
 }
 
+func (c *clientCache) Keys() []ClientCacheKey {
+	var keys []ClientCacheKey
```

Comment:
nit: May as well allocate all the slice memory upfront for efficiency.

```suggestion
	keys := make([]ClientCacheKey, len(c.cache.Keys(), 0)
```